### PR TITLE
Natural sort order in the sidebar

### LINKED ISSUE #855: Numbered files get sorted in an unexpected/non-sequential manner in the sidebar
**Describe the bug**  
Files with sequential numbers in them are sorted in an unexpected manner (sorted by each individual numeral character, not by the whole number's sequential value)

**To Reproduce**  
1. Create some folders/files that span 0-25ish
2. See that the list reads like: 1, 10, 11, 12, ... 19, 2, 20, 21...

**Expected behavior**  
Numbered files should follow in sequential order of the number's value.

**Screenshots**  
image

**Desktop (please complete the following information):**  
 - OS: Fedora 39
 - App Version: Rnote 0.9.0
 - Installation Source: Flatpak/Flathub remote
 - Desktop Environment: Gnome 45

### PR DESCRIPTION
**Pull request by Doublonmousse**
Fixes #855 .

Using `numeric-sort` as the crate for the natural order comparison, in order to have a still maintained crate.

### CODE REVIEW

**Comment by Doublonmousse** [comment]
I only changed the sorting algorithm for sorting files, will also add folders to the mix.

File: crates/rnote-ui/src/workspacebrowser/mod.rs [hunk 1]
```diff
@@ -396,7 +397,7 @@ impl RnWorkspaceBrowser {
             let second_display_name = second_file.basename().unwrap();
             let second_display_name = second_display_name.to_str().unwrap();
 
-            first_display_name.cmp(second_display_name).into()
+            cmp(first_display_name, second_display_name).into()
```
Comment (flxzt) [inline crates/rnote-ui/src/workspacebrowser/mod.rs:400]: let's use `numeric_sort::cmp` explicitly here

Comment (Doublonmousse) [reply crates/rnote-ui/src/workspacebrowser/mod.rs]: fixed

File: crates/rnote-ui/src/workspacebrowser/mod.rs [hunk 2]
```diff
@@ -17,6 +17,7 @@ use gtk4::{
     PropertyExpression, ScrolledWindow, Separator, SignalListItemFactory, SingleSelection,
     SortListModel, SorterChange, Widget,
 };
+use numeric_sort;
```
Comment (flxzt) [inline crates/rnote-ui/src/workspacebrowser/mod.rs:20]: You can remove this import, it's unused

**Comment by flxzt** [comment]
I think you should have rebased on top of or merged with main in order to resolve the conflict

File: crates/rnote-ui/src/workspacebrowser/mod.rs (hunk 2 above)
Comment (Doublonmousse) [reply crates/rnote-ui/src/workspacebrowser/mod.rs]: fixed

**Comment by Doublonmousse** [comment]
Ok, it finally looks reasonable

**Comment by flxzt** [comment]
looks good, thank you!

**Pull request merged**